components/Map.js

```diff
@@ -1,198 +1,513 @@
+            console.error(error);
+            return;
+          }
+          const mostRecentFeature = new Date(
+            featureCollection.features[0].properties.dispatch_ts
+          );
+          const lastUpdate = format(mostRecentFeature, 'MM/YYYY');
+          // set last updated state for this component
+          this.setState({ lastUpdatedDate: lastUpdate });
+          // pass that date to the parent MapContainer component
+          // so we can display that information under the filters
+          this.props.updateDate(this.state.lastUpdatedDate);
         });
-      },
-      onEachFeature: this.bindPopUp,
     });
 
-    this.fatalityFeatureLayer.setWhere(allModesSelected);
+    // Bind pop-ups for crash and fatality points
+    this.map.on('click', e => {
+      // Crash and fatalitiy points are often stacked on top of eachother
+      // in the data. We want the pop-ups to show information about all
+      // the crashes that occured in the clicked location.
 
-    // Add basemap layers and features to map
-    this.map
-      .addLayer(basemapLayer('Gray'))
-      .addLayer(tiledMapLayer({ url: basemap_url }))
-      .addLayer(this.crashFeatureLayer);
+      // We get the number of features at the clicked location
+      const features = this.map.queryRenderedFeatures(e.point);
+      const numFeatures = features.length;
 
-    // Query for feature counts
-    this.updateFeatures(allModesSelected, this.props.dataset);
+      if (!features.length) {
+        return;
+      }
 
-    // Query for last updated date
-    this.crashFeatureLayer
-      .query()
-      .where('1=1')
-      .orderBy('dispatch_ts', 'DESC')
-      .run((error, featureCollection) => {
-        if (error) {
-          // eslint-disable-next-line no-console
-          console.error(error);
-          return;
-        }
-        const mostRecentFeature = new Date(
-          featureCollection.features[0].properties.dispatch_ts
-        );
-        const lastUpdate = format(mostRecentFeature, 'MM/YYYY');
-        // set last updated state for this component
-        this.setState({ lastUpdatedDate: lastUpdate });
-        // pass that date to the parent MapContainer component
-        // so we can display that information under the filters
-        this.props.updateDate(this.state.lastUpdatedDate);
+      // We determine the datefield to use depending on the
+      // selected dataset
+      const dateField =
+        this.props.dataset == 'crash' ? 'dispatch_ts' : 'date_time';
+
+      // We get the mode type and the date of each crash at the location
+      const properties = features
+        .map(feature => [
+          feature.properties.mode_type,
+          feature.properties[dateField],
+        ])
+        // We sort crashes listed in the pop-up by most-recent
+        // to least recent
+        .sort((a, b) => {
+          return a[1] > b[1] ? -1 : 1;
+        });
+
+      new mapboxgl.Popup({ closeOnClickboolean: true })
```

Comment:
Should this just be `closeOnClick: true`?

Reply:
lol yes - thank you